src/jit/gentree.cpp

```diff
@@ -16606,7 +16707,65 @@ CORINFO_CLASS_HANDLE Compiler::gtGetClassHandle(GenTree* tree, bool* isExact, bo
                     *isExact   = false;
                     *isNonNull = false;
                 }
+                else if (base->OperGet() == GT_ADD)
+                {
+                    // This could be a static field access.
+                    //
+                    // See if op1 is a static field base helper call
+                    // and if so, op2 will have the field info.
+                    GenTree* op1 = base->gtOp.gtOp1;
+                    GenTree* op2 = base->gtOp.gtOp2;
+
+                    bool op1IsStaticFieldBase = false;
+
+                    if (op1->OperGet() == GT_CALL)
+                    {
+                        GenTreeCall* op1Call = op1->AsCall();
+
+                        if (op1Call->gtCallType == CT_HELPER)
+                        {
+                            const CorInfoHelpFunc helper = eeGetHelperNum(op1Call->gtCallMethHnd);
+                            switch (helper)
+                            {
+                                // We are looking for a REF type so only need to check for the GC base helpers
+                                case CORINFO_HELP_GETGENERICS_GCSTATIC_BASE:
+                                case CORINFO_HELP_GETSHARED_GCSTATIC_BASE:
+                                case CORINFO_HELP_GETSHARED_GCSTATIC_BASE_NOCTOR:
+                                case CORINFO_HELP_GETSHARED_GCSTATIC_BASE_DYNAMICCLASS:
+                                case CORINFO_HELP_GETGENERICS_GCTHREADSTATIC_BASE:
+                                case CORINFO_HELP_GETSHARED_GCTHREADSTATIC_BASE:
+                                case CORINFO_HELP_GETSHARED_GCTHREADSTATIC_BASE_NOCTOR:
+                                case CORINFO_HELP_GETSHARED_GCTHREADSTATIC_BASE_DYNAMICCLASS:
+                                    op1IsStaticFieldBase = true;
```

Comment:
You could instead add this as a new `HelperCallProperties`  (see util.cpp line 1300-1400)

Reply:
It seems a bit odd to add this sort of thing to helper call properties, since it is a combination of what the helper means and how it is asking. And not all static base helpers qualify (eg R2R helpers do things differently).

Reply:
@briansull Does that make sense?

Reply:
Yes, it is fine how you have it now.